nifi-nar-bundles/nifi-ccda-bundle/nifi-ccda-processors/src/main/java/org/apache/nifi/processors/ccda/ExtractCCDAAttributes.java

```diff
@@ -0,0 +1,383 @@
+     * if true and the document is not valid, then ProcessException will be thrown
+     */
+    public static final PropertyDescriptor SKIP_VALIDATION = new PropertyDescriptor.Builder().name("skip-validation")
+            .displayName("Skip Validation").description("Whether or not to validate CDA message values").required(true)
+            .allowableValues("true", "false").defaultValue("true").addValidator(StandardValidators.BOOLEAN_VALIDATOR)
+            .build();
+
+    /**
+     * PRETTY_PRINTING - Choice of whether generated JSON in the FileFlow content is
+     * pretty printed with new line and tabs or just a continuous string of characters
+     */
+    public static final PropertyDescriptor PRETTY_PRINTING = new PropertyDescriptor.Builder().name("pretty-printing")
+            .displayName("Pretty Printing").description("Whether or not to Pretty Print JSON output").required(true)
+            .allowableValues("true", "false").defaultValue("false").addValidator(StandardValidators.BOOLEAN_VALIDATOR)
+            .build();
+
+    /**
+     * REL_SUCCESS - Value to be returned in case the processor succeeds
+     */
+    public static final Relationship REL_SUCCESS = new Relationship.Builder()
+            .name("success")
+            .description("A FlowFile is routed to this relationship if it is properly parsed as CDA and its content stored in FlowFile")
+            .build();
+
+    /**
+     * REL_FAILURE - Value to be returned in case the processor fails
+     */
+    public static final Relationship REL_FAILURE = new Relationship.Builder()
+            .name("failure")
+            .description("A FlowFile is routed to this relationship if it cannot be parse CDA and store content to FlowFile.")
+            .build();
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return properties;
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return relationships;
+    }
+
+    @Override
+    protected void init(final ProcessorInitializationContext context) {
+        getLogger().info("Loading packages");
+
+        relationships = new HashSet<>();
+        relationships.add(REL_SUCCESS);
+        relationships.add(REL_FAILURE);
+
+        properties = new ArrayList<>();
+        properties.add(SKIP_VALIDATION);
+        properties.add(PRETTY_PRINTING);
+
+        long start = System.currentTimeMillis();
+
+        // Load required MDHT packages
+        System.setProperty( "org.eclipse.emf.ecore.EPackage.Registry.INSTANCE",
+                "org.eclipse.emf.ecore.impl.EPackageRegistryImpl" );
+        CDAPackage.eINSTANCE.eClass();
+        HITSPPackage.eINSTANCE.eClass();
+        CCDPackage.eINSTANCE.eClass();
+        ConsolPackage.eINSTANCE.eClass();
+        IHEPackage.eINSTANCE.eClass();
+        timingStats.add(String.format("Loaded packages in %d ms", System.currentTimeMillis() - start));

```

Comment:
In changing how these run, can you use the logger.info("Message is {}", new Object[]{"value"}) way of doing things. At the slf4j level this is lazy so the strings and calculations aren't computed if it's not at a high enough log level.